tests/cnf/ran/gitopsztp/tests/ztp-bios-day-zero.go

```diff
@@ -0,0 +1,127 @@
+package tests
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/openshift-kni/eco-goinfra/pkg/bmh"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/hive"
+	"github.com/openshift-kni/eco-goinfra/pkg/nodes"
+	"github.com/openshift-kni/eco-goinfra/pkg/reportxml"
+	"github.com/openshift-kni/eco-gotests/tests/cnf/ran/gitopsztp/internal/tsparams"
+	. "github.com/openshift-kni/eco-gotests/tests/cnf/ran/internal/raninittools"
+	"github.com/openshift-kni/eco-gotests/tests/cnf/ran/internal/version"
+	"github.com/openshift-kni/eco-gotests/tests/internal/cluster"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = Describe("ZTP BIOS Configuration Tests", Label(tsparams.LabelBiosDayZeroTests), func() {
+	var (
+		spokeClusterName string
+		nodeNames        []string
+	)
+
+	// 75196 - Check if spoke has required BIOS setting values applied
+	It("Check if SNO spoke has required BIOS setting values applied", reportxml.ID("75196"), func() {
+		versionInRange, err := version.IsVersionStringInRange(RANConfig.ZTPVersion, "4.17", "")
+		Expect(err).ToNot(HaveOccurred(), "Failed to check if ZTP version is in range")
+
+		if !versionInRange {
+			Skip("ZTP BIOS configuration tests require ZTP version of least 4.17")
+		}
+
+		spokeClusterName, err = GetSpokeClusterName(HubAPIClient, Spoke1APIClient)
+		Expect(err).ToNot(HaveOccurred(), "Failed to get SNO cluster name")
+		By(fmt.Sprintf("Cluster name: %s", spokeClusterName))
+
+		nodeNames, err = GetNodeNames(Spoke1APIClient)
+		Expect(err).ToNot(HaveOccurred(), "Failed to get node names")
+		By(fmt.Sprintf("Node names: %v", nodeNames))
+
+		By(fmt.Sprintf("cluster=%s SNO spoke=%s", spokeClusterName, nodeNames[0]))
+		hfs, err := bmh.PullHFS(HubAPIClient, nodeNames[0], spokeClusterName)
+		Expect(err).ToNot(
+			HaveOccurred(),
+			fmt.Sprintf("Failed to get HFS for spoke %s in cluster %s", nodeNames[0], spokeClusterName),
+		)
+
+		hfsObject, err := hfs.Get()
+		Expect(err).ToNot(
+			HaveOccurred(),
+			fmt.Sprintf("Failed to get HFS Obj for spoke %s in cluster %s", nodeNames[0], spokeClusterName),
+		)
+
+		hfsRequstedSettings := hfsObject.Spec.Settings
+		hfsCurrentSettings := hfsObject.Status.Settings
+		allSettingsMatch := true
+		if len(hfsRequstedSettings) > 0 {
```

Comment:
Try using early returns here to avoid nesting, like
```go
if len(hfsRequestedSettings) == 0 {
    Skip("hfs.spec.settings map is empty")
}
```

Should this be a failure instead of a skip though? Does the missing hfsRequestedSettings mean the test doesn't apply to the cluster?

Reply:
Specifying BIOS settings is optional, so an empty hfs.spec.settings isn't necessarily a failure. It does leave it to the user to investigate a Skip when BIOS settings were requested, but they are not showing up in the CR.